### PR TITLE
docs: add docs for getting contract balance

### LINKED ISSUE #916: Add docs for getting a contract's balance
Currently, we have a method to pull a contract's balance:

```typescript
getBalance(assetId: BytesLike) {
  return this.provider.getContractBalance(this.id, assetId);
}
```

But we don't have that documented anywhere; we should.

### PR DESCRIPTION
**Pull request by Torres-ssf**
- Close #916

### CODE REVIEW

**Review by arboleya — COMMENTED** [review]
Would it be useful to showcase the Sway contract that the snippet example interacts with to give users the full picture of what the call below is doing?

```ts
const scope = contract.functions.return_context_amount().callParams({
  forward: [amountToForward, NativeAssetId],
});
```

I did something similar here:

 - https://fuellabs.github.io/fuels-ts/guide/abi-typegen/using-generated-types.html#using-generated-predicate-types

I'm not sure if, in this case, it'd add noise to the `getBalance` stuff, but in general, I have the feeling that it's good to have a full understanding of the examples.

See if you think.

**Review by Dhaiwat10 — DISMISSED** [review]

**Review by danielbate — APPROVED** [review]

File: apps/docs/src/guide/contracts/contract-balance.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+## Checking Contract Balance
+
+Consider a simple contract that transfers a specified amount of a given asset to an address:
+
+<<< @/../../docs-snippets/contracts/transfer-to-address/src/main.sw#contract-balance-2{rust:line-numbers}
+
+The `transfer` function has three parameters:
+
+1. `amount_to_transfer`: The amount that is being transfered.
```
Comment (arboleya) [inline apps/docs/src/guide/contracts/contract-balance.md:19]:
```suggestion
1. `amount_to_transfer`: The amount that is being transferred.
```

File: apps/docs/src/guide/contracts/contract-balance.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+2. `asset`: The address of the deployed contract Token.
+
+3. `recipient`: The address from the receive's wallet.
```
Comment (arboleya) [inline apps/docs/src/guide/contracts/contract-balance.md:23]:
```suggestion
3. `recipient`: The address from the receiver's wallet.
```

File: apps/docs/src/guide/contracts/contract-balance.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+The `transfer` function calls the built-in Sway function `transfer_to_address`, which does exactly what the name suggests.
```
Comment (arboleya) [inline apps/docs/src/guide/contracts/contract-balance.md:25]:
```suggestion
The `transfer` function calls the built-in Sway function `transfer_to_address`, which does precisely what the name suggests.
```

File: apps/docs/src/guide/contracts/contract-balance.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+Let's execute this contract and use the `getBalance` method to validate the remaining asset amount that the contract has left to spend.
```
Comment (arboleya) [inline apps/docs/src/guide/contracts/contract-balance.md:27]:
```suggestion
Let's execute this contract and use the `getBalance` method to validate the remaining asset amount the contract has left to spend.
```

File: apps/docs/src/guide/contracts/contract-balance.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+<<< @/../../docs-snippets/src/guide/contracts/contract-balance.test.ts#contract-balance-3{ts:line-numbers}
+
+In this example, we first forward an asset amount greater than the amount required for the transfer and then execute the contract call.
+
+Finally we use the `getBalance` method to confirm that the contract balance is precisely the total forwarded amount minus the transferred amount.
```
Comment (arboleya) [inline apps/docs/src/guide/contracts/contract-balance.md:33]:
```suggestion
Finally, we use the `getBalance` method to confirm that the contract balance is precisely the total forwarded amount minus the transferred amount.
```

File: apps/docs/src/guide/contracts/contract-balance.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+It is important to note that this method returns the total available contract balance, regardless of how many times assets have been sent or spent on costly operations.
```
Comment (arboleya) [inline apps/docs/src/guide/contracts/contract-balance.md:35]:
```suggestion
It is important to note that this method returns the total available contract balance, regardless of how often assets have been sent or spent on costly operations.
```

**Review by arboleya — CHANGES_REQUESTED** [review]
Very clean and straight to the point. ❤️ 

Left some trivial tyop corrections and suggestions.

File: apps/docs/src/guide/contracts/contract-balance.md (hunk 1 above)
Comment (Torres-ssf) [reply apps/docs/src/guide/contracts/contract-balance.md:19]: Good catch @arboleya

**Review by arboleya — APPROVED** [review]

**Review by Dhaiwat10 — APPROVED** [review]

**Pull request merged**